gateway/tests/integration/createfunction_test.go

```diff
@@ -4,25 +4,114 @@
 package inttests
 
 import (
+	"encoding/json"
 	"net/http"
+	"strings"
 	"testing"
 )
 
-func TestCreate_ValidJson(t *testing.T) {
-	reqBody := `{}`
-	_, code, err := fireRequest("http://localhost:8080/system/functions", http.MethodPost, reqBody)
+type PostFunctionRequest struct {
```

Comment:
My rational behind this was to decouple the integration tests and the request types and also because `CreateFunctionRequest` had fields that aren't used by the UI when creating a function. Are they used by the cli?

Reply:
I figured that was the case, but wondered if it would be easier to maintain if we didn't duplicate the structs in both projects just yet.

Reply:
Sorted